src/model/include/iDynTree/SphericalJoint.h

```diff
@@ -0,0 +1,230 @@
+// SPDX-FileCopyrightText: Fondazione Istituto Italiano di Tecnologia (IIT)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef IDYNTREE_SPHERICAL_JOINT_H
+#define IDYNTREE_SPHERICAL_JOINT_H
+
+#include <iDynTree/Transform.h>
+#include <iDynTree/SpatialMotionVector.h>
+
+#include <iDynTree/Indices.h>
+#include <iDynTree/MovableJointImpl.h>
+#include <iDynTree/VectorFixSize.h>
+
+namespace iDynTree
+{
+    /**
+     * Class representing a spherical joint, i.e. a joint that
+     * constrains two links to have the same position but allows
+     * any relative orientation (3 rotational degrees of freedom).
+     *
+     * The joint state is parameterized using a unit quaternion
+     * q = [qw, qx, qy, qz] (4 position coordinates), and the
+     * angular velocity ω expressed in the first link frame
+     * (3 velocity coordinates).
+     *
+     * Note: The quaternion follows the real/imaginary (w,x,y,z) convention
+     * used by iDynTree, where w is the real part and (x,y,z) is the imaginary part.
+     *
+     * The 6D spatial vectors (velocity, acceleration, forces) follow the
+     * linear/angular (linear first, angular second) convention used by iDynTree.
+     *
+     * \ingroup iDynTreeModel
+     */
+    class SphericalJoint : public MovableJointImpl<4,3>
+    {
+    private:
+        // Structure attributes
+        LinkIndex link1;
+        LinkIndex link2;
+        Transform link1_X_link2_at_rest;
+
+        // Joint center offsets (vector from link origin to joint center, expressed in that link frame)
+        Position m_centerInFirstLink;
+        Position m_centerInSecondLink;
```

Comment:
Not strictly necessary, but I would remove `m_centerInSecondLink`, so we do not need to manually update it every time `m_centerInFirstLink` changes.